lib/Doctrine/ORM/Mapping/UnderscoreNamingStrategy.php

```diff
@@ -40,22 +40,26 @@
  */
 class UnderscoreNamingStrategy implements NamingStrategy
 {
-    private const DEFAULT_PATTERN        = '/(?<=[a-z])([A-Z])/';
-    private const PATTERN_FOR_PROPERTIES = '/(?<=[a-z0-9])([A-Z])/';
+    private const DEFAULT_PATTERN      = '/(?<=[a-z])([A-Z])/';
+    private const NUMBER_AWARE_PATTERN = '/(?<=[a-z0-9])([A-Z])/';
 
     /**
      * @var integer
      */
     private $case;
 
+    /** @var string */
+    private $pattern;
+
     /**
      * Underscore naming strategy construct.
      *
-     * @param integer $case CASE_LOWER | CASE_UPPER
+     * @param int $case CASE_LOWER | CASE_UPPER
      */
-    public function __construct($case = CASE_LOWER)
+    public function __construct($case = CASE_LOWER, bool $numberAware = false)
```

Comment:
We should probably deprecate not setting this to `true` and warn our users that the class will behave as if it were set to true in the next major version

Reply:
Yeap, planned for `2.7.0` didn't want to add deprecations in a patch.